### PR TITLE
[MG Framework] Implement modifier keys API

### PR DESCRIPTION
**Pull request by cra0zy**
This adds an API that exposes CapsLock / NumLock states to the end user (#4729). I didn't add scroll lock because SDL doesn't expose it and looking into how to add it seemed pointless... when was the last time someone even used it?

### CODE REVIEW

**Comment by tomspilman** [comment]
Glad to see the cleanup of the `Keyboard` classes.

As long as we can correctly report these states on all platforms i'm good with the addition.  It just has to be reliable else it is a waste of time to add it.

@KonajuGames @dellis1972 ?

File: MonoGame.Framework/Input/Keyboard.SDL.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+namespace Microsoft.Xna.Framework.Input
+{
+    public static partial class Keyboard
+    {
+        static List<Keys> _keys;
+
+        private static KeyboardState PlatformGetState()
+        {
+            var modifiers = Sdl.Keyboard.GetModState();
+            return new KeyboardState(_keys, modifiers.HasFlag(Sdl.Keyboard.Keymod.CapsLock), modifiers.HasFlag(Sdl.Keyboard.Keymod.NumLock));
```
Comment (KonajuGames) [inline MonoGame.Framework/Input/Keyboard.SDL.cs]: Enum.HasFlag() is slower and can generate garbage as it does [type checks](http://referencesource.microsoft.com/#mscorlib/system/enum.cs,877).  I would prefer to see a simpler and garbage free test such as

    (modifiers & Sdl.Keyboard.Keymod.CapsLock) == Sdl.Keyboard.Keymod.CapsLock

Comment (cra0zy) [inline MonoGame.Framework/Input/Keyboard.SDL.cs]: Done.

**Comment by tomspilman** [comment]
Some unit tests for keyboard would help with https://github.com/mono/MonoGame/issues/4744.

**Comment by cra0zy** [comment]
> Some unit tests for keyboard would help with #4744.

You're evil :P

**Comment by tomspilman** [comment]
> You're evil :P

Someone has to be the unit test bad guy. :)

**Pull request merged**